platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/TelemetryImpl.java

```diff
@@ -0,0 +1,158 @@
+    MapState state = new MapState(latitude, longitude, zoom);
+    state.setGesture(eventType);
+    telemetry.push(mapEventFactory.createMapGestureEvent(Event.Type.MAP_CLICK, state));
+  }
+
+  /**
+   * Set the end-user selected state to participate or opt-out in telemetry collection.
+   */
+  @Override
+  public void setUserTelemetryRequestState(boolean enabledTelemetry) {
+    if (enabledTelemetry) {
+      TelemetryEnabler.updateTelemetryState(TelemetryEnabler.State.ENABLED);
+      telemetry.enable();
+    } else {
+      telemetry.disable();
+      TelemetryEnabler.updateTelemetryState(TelemetryEnabler.State.DISABLED);
+    }
+  }
+
+  /**
+   * Set the debug logging state of telemetry.
+   *
+   * @param debugLoggingEnabled true to enable logging
+   */
+  @Override
+  public void setDebugLoggingEnabled(boolean debugLoggingEnabled) {
+    telemetry.updateDebugLoggingEnabled(debugLoggingEnabled);
+  }
+
+  /**
+   * Set the telemetry rotation session id interval
+   *
+   * @param interval the selected session interval
+   * @return true if rotation session id was updated
+   */
+  @Override
+  public boolean setSessionIdRotationInterval(int interval) {
+    return telemetry.updateSessionIdRotationInterval(new SessionInterval(interval));
+  }
+
+  /**
+   * Set the debug logging state of telemetry.
+   *
+   * @param debugLoggingEnabled true to enable logging
+   * @deprecated use {@link #setDebugLoggingEnabled(boolean)} instead
+   */
+  @Deprecated
+  public static void updateDebugLoggingEnabled(boolean debugLoggingEnabled) {
+    getInstance().setDebugLoggingEnabled(debugLoggingEnabled);
+  }
+
+  /**
+   * Update the telemetry rotation session id interval
+   *
+   * @param interval the selected session interval
+   * @return true if rotation session id was updated
+   * @deprecated use {@link #setSessionIdRotationInterval(int)} instead
+   */
+  @Deprecated
+  public static boolean updateSessionIdRotationInterval(SessionInterval interval) {
+    try {
+      Field field = interval.getClass().getDeclaredField("interval");
+      field.setAccessible(true);
+      Integer intervalValue = (Integer) field.get(interval);
+      return getInstance().setSessionIdRotationInterval(intervalValue);
+    } catch (Exception exception) {
+      return false;
+    }
+  }
+
+  /**
+   * Method to be called when an end-user has selected to participate in telemetry collection.
+   *
+   * @deprecated use {@link #setUserTelemetryRequestState(boolean)} with parameter true instead
+   */
+  @Deprecated
+  public static void enableOnUserRequest() {
+    getInstance().setUserTelemetryRequestState(true);
+  }
+
+  /**
+   * Method to be called when an end-user has selected to opt-out of telemetry collection.
+   *
+   * @deprecated use {@link #setUserTelemetryRequestState(boolean)} with parameter false instead
+   */
+  @Deprecated
+  public static void disableOnUserRequest() {
+    getInstance().setUserTelemetryRequestState(false);
```

Comment:
This method should get the `TelemetryImpl` reference using the `Mapbox#getTelemetry` and check for nullability to avoid crashing.